### PR TITLE
[CPM-80] Add regulation generation into courses

### PR DESCRIPTION
**Pull request by daltonlim**

### CODE REVIEW

**Comment by FocalChord** [comment]
Could we fill in corequisites and informal equivalents too please?

**Comment by daltonlim** [comment]
> Could we fill in corequisites and informal equivalents too please?
Logic for randomly assigning that will take a while to figure out. Are we better off just modelling an existing programme and using the same data every time?

**Comment by FocalChord** [comment]
> > Could we fill in corequisites and informal equivalents too please?
> 
> Logic for randomly assigning that will take a while to figure out. Are we better off just modelling an existing programme and using the same data every time?

Yup let's build an existing programme and use that data for now. It's probably better than having no data.

**Comment by daltonlim** [comment]
> > > Could we fill in corequisites and informal equivalents too please?
> > 
> > 
> > Logic for randomly assigning that will take a while to figure out. Are we better off just modelling an existing programme and using the same data every time?
> 
> Yup let's build an existing programme and use that data for now. It's probably better than having no data.

Yea nvm. turns out that i just needed to increase database size. The first 20 have different restrictions, prereqs, coreqs and informal equivalents and all other courses have nothing.

File: seed-db/scripts/course.js [hunk 1]
```diff
@@ -25,18 +26,35 @@ const generateCourseList = () => {
...
         semester: getRandomElement(semesters),
-        restrictions: [],
-        prerequisites: [],
-        corequisites: [],
+        restrictions: i <20 ? getRandomElement(restrictions) : [],
+        prerequisites: i< 20 ? getRandomElement(prerequisites): [],
+        corequisites: i< 20 ? getRandomElement(coRequisites): [],
         description: faker.lorem.paragraph(),
-        informalEquivalents: []
+        informalEquivalents: i< 20 ? [getCourseID(Math.floor(Math.random() * 20) + 21)]: [],
```
Comment (Richardng12) [inline seed-db/scripts/course.js:57]: Are these all just like generating random courses?

File: seed-db/scripts/regulation.js [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
-const points = [45,60,100]
+const points = [15]
 const pointRequirement = ["UPTO", "EXACT", "ATLEAST"]
 
 const createRegulation = (i) => {
     const regulation = {
         points: getRandomElement(points),
         pointRequirement: getRandomElement(pointRequirement),
-        courses: []
+        courses: [i >= 81 ? getCourseID(Math.floor(Math.random() * 20) + 81) : [] ||
```
Comment (Richardng12) [inline seed-db/scripts/regulation.js:13]: random wizardry?

File: seed-db/scripts/course.js (hunk 1 above)
Comment (daltonlim) [reply seed-db/scripts/course.js:57]: Yeah just picks a random course id

**Review by Richardng12 — APPROVED** [review]
lgtm wizard

File: seed-db/scripts/index.js [hunk 3]
```diff
@@ -6,7 +6,6 @@ const { generateRegulations } = require("./regulation");
 const { generateProgrammeDegrees } = require("./programme-degree");
```
Comment (wryao64) [inline seed-db/scripts/index.js:6]: Line 5 needs to be removed - docker-compose gives an error because regulation.js doesn't exist

**Review by wryao64 — CHANGES_REQUESTED** [review]
Could you also please comment out the logger print statement that prints all the courses to the console?

File: seed-db/scripts/index.js (hunk 3 above)
Comment (daltonlim) [reply seed-db/scripts/index.js:5]: done

**Review by wryao64 — APPROVED** [review]

**Pull request merged**